bindingcollectionadapter/src/main/java/me/tatarka/bindingcollectionadapter2/itembindings/OnItemBindClass.java

```diff
@@ -18,18 +21,30 @@
  */
 public class OnItemBindClass<T> implements OnItemBind<T> {
 
-    private final SimpleArrayMap<Class<? extends T>, int[]> itemBindingMap;
+    private final SimpleArrayMap<Class<? extends T>, Object> itemBindingMap;
+    // to support ordinal mapping {@see OnItemBindClassTest#selectsBasedOnClassAndSubclass()}
```

Comment:
I came to the conclusion that I should use [LinkedHashMap](http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html) to prevent duplicates of mappings. And it supports insertion ordering.